frontend/src/Quiz/Quiz.vue

```diff
@@ -0,0 +1,453 @@
+<script setup lang="ts">
+/**
+ * This component displays a quiz that can "work" in 4 modes, based on passed parameters.
+ * It is available for teachers and students.
+ *
+ * For teacher, it can act like a component to view detail of quiz, or to score quiz.
+ * For student, it can act like a component to fill a quiz, or to view quiz results.
+ */
+import { ref, onMounted, onUnmounted } from 'vue';
+import VueCountdown from '@chenfengyuan/vue-countdown';
+import { getDataWithCSRF } from '../utilities/api';
+
+const { quiz_html, is_teacher, student, enrolled_id, answers, remaining, scoring, scored_by } =
+  defineProps<{
+    quiz_html: string;
+    user: number;
+    student: string;
+    remaining: string;
+    answers: string;
+    scoring: string;
+    quiz_id: number;
+    is_teacher: string;
+    enrolled_id: number;
+    scored_by: string;
+  }>();
+
+enum Modes {
+  Display,
+  Filling,
+  Scoring,
+  ViewScoring
+}
+
+enum ResultsType {
```

Comment:
I guess that this represents something like `SubmitType`? And the values should be `Periodic` and `Manual/Final`.